src/overlays/actors/ovl_Boss_Tw/z_boss_tw.c

```diff
@@ -3411,45 +3411,48 @@ void func_80942C70(Actor* thisx, PlayState* play) {
+        gDPSetEnvColor(POLY_XLU_DISP++, 255, 0, 0, 128);
+    } else {
+        gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 255, 255, 255, alpha);
+        gDPSetEnvColor(POLY_XLU_DISP++, 100, 100, 255, 128);
+    }
+
+    Matrix_Translate(this->beamOrigin.x, this->beamOrigin.y, this->beamOrigin.z, MTXMODE_NEW);
+    Matrix_RotateY(this->beamYaw, MTXMODE_APPLY);
+    Matrix_RotateX(this->beamPitch, MTXMODE_APPLY);
+    Matrix_RotateZ(this->beamRoll, MTXMODE_APPLY);
+    Matrix_Scale(this->beamScale, this->beamScale, (this->beamDist * 0.01f * 98.0f) / 20000.0f, MTXMODE_APPLY);
+    gSPMatrix(POLY_XLU_DISP++, MATRIX_NEW(play->state.gfxCtx, "../z_boss_tw.c", 6846),
+              G_MTX_LOAD | G_MTX_MODELVIEW | G_MTX_NOPUSH);
+    gSPDisplayList(POLY_XLU_DISP++, SEGMENTED_TO_VIRTUAL(gTwinrovaBeamDL));
 
-        Matrix_Translate(this->beamOrigin.x, this->beamOrigin.y, this->beamOrigin.z, MTXMODE_NEW);
-        Matrix_RotateY(this->beamYaw, MTXMODE_APPLY);
-        Matrix_RotateX(this->beamPitch, MTXMODE_APPLY);
+    if (this->beamReflectionDist > 10.0f) {
+        Matrix_Translate(this->beamReflectionOrigin.x, this->beamReflectionOrigin.y, this->beamReflectionOrigin.z,
+                         MTXMODE_NEW);
+        Matrix_RotateY(this->beamReflectionYaw, MTXMODE_APPLY);
+        Matrix_RotateX(this->beamReflectionPitch, MTXMODE_APPLY);
         Matrix_RotateZ(this->beamRoll, MTXMODE_APPLY);
-        Matrix_Scale(this->beamScale, this->beamScale, (this->beamDist * 0.01f * 98.0f) / 20000.0f, MTXMODE_APPLY);
-        gSPMatrix(POLY_XLU_DISP++, MATRIX_NEW(play->state.gfxCtx, "../z_boss_tw.c", 6846),
+        Matrix_Scale(this->beamScale, this->beamScale, (this->beamReflectionDist * 0.01f * 100.0f) / 20000.0f,
+                     MTXMODE_APPLY);
+        gSPMatrix(POLY_XLU_DISP++, MATRIX_NEW(play->state.gfxCtx, "../z_boss_tw.c", 6870),
                   G_MTX_LOAD | G_MTX_MODELVIEW | G_MTX_NOPUSH);
         gSPDisplayList(POLY_XLU_DISP++, SEGMENTED_TO_VIRTUAL(gTwinrovaBeamDL));
-
-        if (this->beamReflectionDist > 10.0f) {
-            Matrix_Translate(this->beamReflectionOrigin.x, this->beamReflectionOrigin.y, this->beamReflectionOrigin.z,
-                             MTXMODE_NEW);
-            Matrix_RotateY(this->beamReflectionYaw, MTXMODE_APPLY);
-            Matrix_RotateX(this->beamReflectionPitch, MTXMODE_APPLY);
-            Matrix_RotateZ(this->beamRoll, MTXMODE_APPLY);
-            Matrix_Scale(this->beamScale, this->beamScale, (this->beamReflectionDist * 0.01f * 100.0f) / 20000.0f,
-                         MTXMODE_APPLY);
-            gSPMatrix(POLY_XLU_DISP++, MATRIX_NEW(play->state.gfxCtx, "../z_boss_tw.c", 6870),
-                      G_MTX_LOAD | G_MTX_MODELVIEW | G_MTX_NOPUSH);
-            gSPDisplayList(POLY_XLU_DISP++, SEGMENTED_TO_VIRTUAL(gTwinrovaBeamDL));
-        }
-
-        Matrix_Pop();
     }
 
+    Matrix_Pop();
+
+end:
     CLOSE_DISPS(play->state.gfxCtx, "../z_boss_tw.c", 6878);
```

Comment:
theres a single case of a draw function end label being called close_disps, and I kind of like it. it lets you know from the goto that its gonna do end disps like it should and then finish. 
That said there are way more existing cases of "end". So we can decide and clean it up some other time

Reply:
there's only 2 other cases of this so I went ahead and changed those too